### DISCUSSION TITLE
Best practice for backups on BTRFS

### DISCUSSION THREAD

**Discussion by lucmos** [creativeprojects/resticprofile#364] [Q&A]
Hello, 
I would like to port my hacky scripts to perform backups with restic to resticprofile :smile: 

They work in the following way:

```bash
#!/usr/bin/bash

btrfs subvolume snapshot -r /srv/dev-disk-by-label-data/luca/ /srv/dev-disk-by-label-data/.snapshots/srv/dev-disk-by-label-data/luca

echo -e "\n\n~~~~~\n\n"

. /root/restic_envs_passport.sh
echo -e "\n\n\n---\nBACKUP TO: ${RESTIC_REPOSITORY}\n---\n"
/root/bin/restic --verbose backup "/srv/dev-disk-by-label-data/.snapshots/srv/dev-disk-by-label-data" --tag data --exclude-file="/root/backup-exclude.txt"
echo -e "\n---\nDIFF\n---\n"
/root/bin/restic snapshots --json | python -c "import sys, json; j=json.load(sys.stdin); j = [x for x in j if 'tags' not in x or 'fireflyiii' not in x['tags'] or 'actual' not in x['tags']]; print(j[-2]['short_id'] + ' ' + j[-1]['short_id'])" | xargs /root/bin/restic diff

echo -e "\n\n~~~~~\n\n"

btrfs subvolume delete /srv/dev-disk-by-label-data/.snapshots/srv/dev-disk-by-label-data/luca
```

The idea is the following:

1. The backup scripts first creates a btrfs snapshots, to freeze the state and avoid inconsistent backups (in case the state changes while the backup is in progress. A classic example is backing up a live database, that often is unreadable without this kind of state freeze)
2. Execute restic
3. Print the differences with an hacky one liner. This is useful because, since it's executed in a crontab, these differences are emailed directly to me
4. Remove the btrfs temporary snapshots

I am interested in automating all this with restic profile. 
Could you give some pointer on what is the best way to do it?

My current thinking was to use the [commands hooks](https://creativeprojects.github.io/resticprofile/configuration/run_hooks/index.html). Is that the best way to handle the btrfs snapshots or should they be handled differently?

Instead, I have no idea on how to track the changes via email

Any help is much appreciated! :smile:

**Comment by creativeprojects** [comment]
Hey!

Sorry I just realised I missed your query 😞 

Your thinking about using command hooks is right, that's exactly how I would do it 😄 

About sending an email on error, you can use the automated scheduler with `cron`:  https://creativeprojects.github.io/resticprofile/schedules/index.html.

You can also send emails on error with this trick using `systemd`: https://github.com/creativeprojects/resticprofile/tree/master/contrib/systemd
